packages/neos-ui-editors/src/Editors/Reference/referenceDataLoader.js

```diff
@@ -103,9 +104,13 @@ export default ({isMulti}) => WrappedComponent => {
         }
 
         getDataLoaderOptions() {
+            const contextNode = this.props.options.startingPoint ?
+                this.props.contextForNodeLinking.set('contextNode', this.props.options.startingPoint) :
```

Comment:
you are modifying props there. I don't think that should EVER happen. You might need to create a deep clone of contextForNodeLinking if starting point is set and then set it on that copy.

Reply:
Does it? contextForNodeLinking is a Immutable map and i tought that the set() creates a copy (https://facebook.github.io/immutable-js/ second code example)

Reply:
It's initalizied here: https://github.com/neos/neos-ui/blob/master/packages/neos-ui-redux-store/src/UI/NodeLinking/index.js#L24

Reply:
Oh, is it? In that case YAY, but IMHO the wording is awkward, not your fault though.. `set` shouldn't do `createDeepCloneWithThisPropertyChanged`